app/src/main/java/com/vasilevkin/catsanddogs/repository/AnimalRepository.kt

```diff
@@ -0,0 +1,48 @@
+package com.vasilevkin.catsanddogs.repository
+
+import com.vasilevkin.catsanddogs.models.localModels.Animal
+import com.vasilevkin.catsanddogs.repository.dataSource.ICloudDataSource
+import io.reactivex.Single
+
+
+class AnimalRepository(
+    private val cloudDataSource: ICloudDataSource
+) : IAnimalRepository {
+
+    override fun getAllAnimals(): Single<List<Animal>> {
+        return cloudDataSource.getAnimals().map {
+                it.map {
+                    Animal(null, it.photo[0].orEmpty())
```

Comment:
comment above about defauld parameter - Animal(imageUrl = it.photo.first().orEmpty())

Reply:
Please explain in more details.

Reply:
There above we created Animal with default param "val title: String? = null", after that you can init your Animal object like "Animal(imageUrl = it.photo.first().orEmpty())" without first param because this param has default value

Reply:
Thanks! 
Fixed.